Fix bad URI regex formatting in caseologue.py

hmenager opened the pull request:
fix #25

hmenager committed:
Fix bad URI regex formatting in caseologue.py

fix #25

rioualen reviewed (approved):
Looks fine to me, thank you for fixing it!

matuskalas commented on caseologue_python/caseologue.py:
```diff
@@ -228,7 +228,7 @@ def test_bad_uri(self):
                         "bad_rui",
```
I guess this wanted to be "bad_uri"? 😊

matuskalas replied:
```suggestion
                        "bad_uri",
```

matuskalas reviewed (commented):
Many thanks for fixing!

It is very obviously fixed here
https://github.com/edamontology/caseologue/actions/runs/17649216086/job/50155415591
as opposed to
https://github.com/edamontology/caseologue/actions/runs/17669817538/job/50218902402
I'm just wondering here, is there some concurrency, or why is the "Ran 13 tests ..." output before versus after the table in those 2 examples? :O

matuskalas reviewed (approved):
This test is nicely tested here
https://github.com/edamontology/caseologue/actions/runs/17649215970/job/50155403817
~~And all is great 👍🏽
So I think we can merge 🙌🏽~~

I'm just puzzled by the way is was (not) failing before, see https://github.com/edamontology/caseologue/actions/runs/17672367576/job/50226726471
~~Isn't that a bug in the test that it actually passed as ok? (compared to the fixed test above)~~

matuskalas commented:
> I'm just puzzled by the way is was (not) failing before, see https://github.com/edamontology/caseologue/actions/runs/17672367576/job/50226726471 Isn't that a bug in the test that it actually passed as ok? (compared to the fixed test above)

I see, it just prints the regex strings in exactly the same way in both cases.

matuskalas commented on caseologue_python/caseologue.py:
```diff
@@ -228,7 +228,7 @@ def test_bad_uri(self):
                         r["entity"],
                         f"'{r['label']}'",
-                        "has a bad URI (entity) (regex :^http://edamontology.org/(data|topic|operation|format)_[0-9]\{4\}$)",
+                        r"has a bad URI (entity) (regex :^http://edamontology.org/(data|topic|operation|format)_[0-9]\{4\}$)",
```
Do you think it would cause a problem if it was just
```suggestion
                        r"has a bad URI (entity) (regex :^http://edamontology.org/(data|topic|operation|format)_[0-9]{4}$)",
```
?  (sorry for bugging, no time to try it myself in Python right now, need to prepare a short talk in <30min)

matuskalas replied:
Otherwise maybe
```suggestion
                        r"has a bad URI (entity) (regex :^http://edamontology.org/(data|topic|operation|format)_[0-9]{{4}}$)",
```
??

matuskalas replied:
Sorry for all the guesses, now I looked it up and tested. The correct string is
```suggestion
                        "has a bad URI (entity) (regex :^http://edamontology.org/(data|topic|operation|format)_[0-9]{4}$)",
```
There is no need to escape {}, therefore the warning. And there are no backslashes we'd need to retain (those would either need `"\\1"` or `r"\1"`), therefore the string had the same problem as before.

If you want to see a lot of black screen:
<img width="996" height="583" alt="image" src="https://github.com/user-attachments/assets/3e3f4408-989c-480b-ba4e-cc8fd52e5950" />

matuskalas committed:
Applied suggestions from code review

These are just strings to print in the standard output of the validator...
- fixed "bad_rui" (uri)
- fixed the regex string

matuskalas commented:
Now the test output looks better
https://github.com/edamontology/caseologue/actions/runs/17679440926/job/50249787457

And the test of EDAM is also without the warning
https://github.com/edamontology/caseologue/actions/runs/17679440988/job/50249800931?pr=26

albangaignard reviewed (approved):
Looks good !
